packages/dev/core/src/FlowGraph/Blocks/Data/Math/flowGraphMathBlocks.ts

```diff
@@ -41,7 +62,7 @@ export class FlowGraphAddBlock extends FlowGraphBinaryOperationBlock<any, any, a
     private _polymorphicAdd(a: any, b: any) {
         const aClassName = _getClassNameOf(a);
         const bClassName = _getClassNameOf(b);
-        if (_areSameVectorClass(aClassName, bClassName)) {
+        if (_areSameVectorClass(aClassName, bClassName) || _areSameMatrixClass(aClassName, bClassName) || _areSameIntegerClass(aClassName, bClassName)) {

```

Comment:
only if you think it makes sense for future usage - this can be a single helper function (something like "notBaseType" or something like that). Feel free to resolve this!

Reply:
I'll resolve since this construction is only used in two places - if we need to add more then I can add the notBaseType :D